src/node/desktop/src/main/detect-r.ts

```diff
@@ -278,29 +278,30 @@ function scanForR(): Expected<string> {
 }
 
 function scanForRPosix(): Expected<string> {
-  // first, look for R on the PATH
+  if (process.platform == 'darwin') {
+    // For Mac, we want to look in a list of hard-coded locations
+    const defaultLocations = ['/opt/local/bin/R', '/usr/local/bin/R', '/usr/bin/R'];
+    // also check framework directory and homebrew ARM locations for macOS
+    defaultLocations.push('/Library/Frameworks/R.framework/Resources/bin/R');
+    defaultLocations.push('/opt/homebrew/bin/R');
+
+    for (const location of defaultLocations) {
+      if (isValidBinary(location)) {
+        logger().logDebug(`Using ${location} (found by searching known locations)`);
+        return ok(location);
+      }
+    }
+  }
+
+  // look for R on the PATH
+  // should we launch the default shell to pick up the user modifications to the path?
   const [rLocation, error] = executeCommand('/usr/bin/which R');
   if (!error && rLocation) {
+    logger().logDebug(`PATH: ${getenv("PATH")}`);
     logger().logDebug(`Using ${rLocation} (found by /usr/bin/which/R)`);
     return ok(rLocation);
   }
 
-  // otherwise, look in some hard-coded locations
-  const defaultLocations = ['/opt/local/bin/R', '/usr/local/bin/R', '/usr/bin/R'];
```

Comment:
https://github.com/rstudio/rstudio/blob/8df7e8c7c6b7727779786613a9089ba14e0c4a68/src/cpp/core/r_util/REnvironmentPosix.cpp#L228-L229

Removing this means that Linux has no fallback locations to check.

The priority order is definitely a bit weird. Linux prefers using `which R` first with a fallback location list and Mac only uses a defined location list.

I think making the location list match the platforms in Qt then adding a platform check for whether to use `which R` will give the same behaviour.